ert_shared/status/tracker/legacy.py

```diff
@@ -221,7 +221,11 @@ def _set_iter_queue(self, iter_: int, queue: JobQueue) -> None:
             self._iter_queue[iter_] = None
 
         if self._iter_queue[iter_] is None and queue is not None:
-            self._iter_queue[iter_] = queue
+            # This function might be called when the iteration on the run model
+            # has been updated, but the queue has yet to be updated. Hence we check that a
+            # queue with the same hash is not found in the map before setting it.
+            if hash(queue) not in {hash(q) for q in self._iter_queue.values()}:
```

Comment:
Performance nitpick! Prefer using generators in this case:
```py
hash(queue) not in (hash(q) for q in self._iter_queue.values())
```

The set version first creates the entire set by for-looping over `self._iter_queue.values()`, while the generator version tests each element by itself. The generator exits early when it finds something too.

Compare:
```py
>>> 2 in {2*x for x in range(10**7)}
>>> 2 in (2*x for x in range(10**7))  # Very quick
>>> 1 in {2*x for x in range(10**7)}
>>> 1 in (2*x for x in range(10**7))
```